autoload/elixir/indent.vim

```diff
@@ -62,6 +62,14 @@ function! elixir#indent#deindent_ending_symbols(ind, line)
 endfunction
 
 function! elixir#indent#deindent_keywords(ind, line)
+  echom "> pair start ". s:PAIR_START
+  echom "> pair middle ". s:PAIR_MIDDLE
+  echom "> pair end ". s:PAIR_END
+  echom "> pair skip ". s:PAIR_SKIP
+
+  " NOTE: cursor is not being properly set here which causes this to
+  " incorrectly handle nesting
+  echom "> cursor => " . line(".") ", " . col(".")
   if a:line.current.text =~ s:DEINDENT_KEYWORDS
     let bslnum = searchpair(
```

Comment:
@jbodah I was trying to understand why the `searchpair` is not working, and my findings are:
1. the `\zs` at the end of the `s:PAIR_END` is what make these tests break, however, if we remove it the other tests will break.
2. Also find another case where the indentation will fail:
```elixir
(:foo) ->
  :foo # fn
  :asd
end
```

With that in mind I think that we'll need to make this function check more details:
1. Instead of `if a:line.current.text =~ s:DEINDENT_KEYWORDS` we'll need something like `if has_indentable_match(a:line.current, s:DEINDENT_KEYWORDS)`
2. Instead of `searchpair` I think we'll need to use the `searchpairpos` and check if the given `pos` is indentable
3. Understand better the reason of the `\zs` (I don't remember, I just know that it's related to indent the `end` word while typing.)